src/redux/preview.js

```diff
@@ -333,26 +381,90 @@ module.exports.getRemixes = id => (dispatch => {
     });
 });
 
-module.exports.getStudios = id => (dispatch => {
-    dispatch(module.exports.setFetchStatus('studios', module.exports.Status.FETCHING));
+module.exports.getProjectStudios = id => (dispatch => {
+    dispatch(module.exports.setFetchStatus('projectStudios', module.exports.Status.FETCHING));
+    api({
+        uri: `/projects/${id}/studios`
+    }, (err, body, res) => {
+        if (err) {
+            dispatch(module.exports.setFetchStatus('projectStudios', module.exports.Status.ERROR));
+            dispatch(module.exports.setError(err));
+            return;
+        }
+        if (typeof body === 'undefined') {
+            dispatch(module.exports.setFetchStatus('projectStudios', module.exports.Status.ERROR));
+            dispatch(module.exports.setError('No projectStudios info'));
+            return;
+        }
+        if (res.statusCode === 404) { // NotFound
+            body = [];
+        }
+        dispatch(module.exports.setFetchStatus('projectStudios', module.exports.Status.FETCHED));
+        dispatch(module.exports.setProjectStudios(body));
+    });
+});
+
+module.exports.getCuratedStudios = (username, token) => (dispatch => {
+    dispatch(module.exports.setFetchStatus('curatedStudios', module.exports.Status.FETCHING));
     api({
-        uri: `/projects/${id}/studios?limit=5`
+        uri: `/users/${username}/studios/curate`,
+        authentication: token
```

Comment:
Is this line necessary? It's just for getting the studios a user curates, which does not require an authentication token ([see this, for example](https://api.scratch.mit.edu/users/mres/studios/curate)).

Reply:
Looks like you're right! Fixed.

Reply:
Also, good catch! It's hard to notice that I'm mistakenly providing too much info to the API.